Test for Payment and Hosted Payment

Jeshan7 opened the pull request:
## Why?
  In payment.ex (model)
   - Test cases for get/1, get_all/0 and from_payment/1 were not present.
  In hosted_payment.ex (model)
   - Test cases for get_all/0 and delete/1 were not present.

## This change addresses the need by:
 In payment.ex (model)
   - Test cases for get/1, get_all/0 and from_payment/1 are now present.
 In hosted_payment.ex (model)
   - Test cases for get_all/0 and delete/1 are now present.

 Refactored some lines of payment_test.exs and hosted_payment_test.exs

[delivers #163829106]

## Checklist
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [ ] I have read [CONTRIBUTING.md][contributing].
- [ ] My code follows the [style guidelines][contributing] of this project.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] My code does not generate any (new) [`credo`][credo] and compile-time warnings.
- [ ] I have updated the documentation wherever necessary.
- [ ] I have added tests to cover my changes.

[contributing]: https://github.com/aviabird/snitch/CONTRIBUTING.md
[credo]: https://github.com/rrrene/credo

Jeshan7 committed:
[#163829106]

Prior to this change,
  In payment.ex (model)
   - Test cases for get/1, get_all/0 and from_payment/1 were not present.
  In hosted_payment.ex (model)
   - Test cases for get_all/0 and delete/1 were not present.
This change
 In payment.ex (model)
   - Test cases for get/1, get_all/0 and from_payment/1 are now present.
 In hosted_payment.ex (model)
   - Test cases for get_all/0 and delete/1 are now present.

 Refactored some lines of payment_test.exs and hosted_payment_test.exs

arjun289 commented on apps/snitch_core/test/data/model/payment/hosted_payment_test.exs:
```diff
@@ -52,6 +52,34 @@ defmodule Snitch.Data.Model.HostedPaymentTest do
+      assert {:error, :hosted_payment_not_found} = HostedPayment.get(-1)
+    end
+
+    test "returns a hosted payment with  map", %{hosted_payment: hosted_payment} do
```
`map` -> `query params`

arjun289 reviewed (commented):
- Please add tests for the error case for `create` and `update` in `hosted_payment` model.

arjun289 commented on apps/snitch_core/test/data/model/payment/payment_test.exs:
```diff
@@ -37,11 +34,21 @@ defmodule Snitch.Data.Model.PaymentTest do
   end
 
+  test "get_all/0 return all payments", %{ccd: cod, chk: check, hpm: hosted_payment} do
+    returned_hosted_payments = Payment.get_all()
```
variable name should be `returned_payments` and not `returned_hosted_payments`, since it returns payments

Jeshan7 committed:
Merge branch 'develop' into payment_tests

Jeshan7 committed:
[#163829106]

Prior to this change,
  - No test cases were present for create/5 and update/3 in
    hosted_payment(model).

This change
  - Wrote new test cases for create/5 and update/3 in
    hosted_payment(model).
  - Changed map -> query params in a test statement of get/1
  - Changed returned_hosted_payments -> returned_payments.

arjun289 commented on apps/snitch_core/test/data/model/payment/hosted_payment_test.exs:
```diff
@@ -17,39 +17,153 @@ defmodule Snitch.Data.Model.HostedPaymentTest do
+      assert %{slug: ["has already been taken"]} == errors_on(changeset)
+    end
+
+    test "" do
```
no statement

Jeshan7 committed:
[#163829106]

Prior to this change,
 - Unnecessary test case was present in create/5.

This change
 -  Deleted the test case in create/5

ashish173 committed:
Merge branch 'develop' into payment_tests
